### PR TITLE
added Character Setup

### LINKED ISSUE #134: Character Preferences
What it says on the thin. Should be stored client side in the user data directories so that we don't have to store them server side. Client's problems.

### PR DESCRIPTION
**Pull request by DamianX**
Closes #134

Provides the UI to create, delete, edit character profiles
There's a half-assed preview of the selected character in the lobby
The preferences are applied to the mob that spawns when you enter the game
Changed HumanoidCharacterProfile and HumanoidCharacterAppearance from classes to record-like immutable structs

~~Apparently letting rider auto-clean committed files was a terrible idea and now I have a ton of unrelated changes in here, which I might now revert maybe possibly if it's important?~~ Fixed.

### CODE REVIEW

File: Content.Client/UserInterface/CharacterSetupGui.cs [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        private class CharacterPickerButton : Control
+        {
+            public readonly Button ActualButton;
+
+            public CharacterPickerButton(
+                IEntityManager entityManager,
+                IClientPreferencesManager preferencesManager,
+                ButtonGroup group,
+                ICharacterProfile profile,
+                string slotName,
```
Comment (DamianX) [inline Content.Client/UserInterface/CharacterSetupGui.cs]: I need to change this method to read these 3 values from `profile` instead, refactoring mishap

File: Content.Client/GameObjects/Components/Mobs/LooksComponent.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
+using Content.Shared.GameObjects.Components.Mobs;
+using Content.Shared.Preferences;
+using Content.Shared.Preferences.Appearance;
+using Robust.Client.GameObjects;
+using Robust.Shared.GameObjects;
+
+namespace Content.Client.GameObjects.Components.Mobs
+{
+    [RegisterComponent]
+    public sealed class LooksComponent : SharedLooksComponent
```
Comment (DamianX) [inline Content.Client/GameObjects/Components/Mobs/LooksComponent.cs]: I hate this name, but Appearance exists and it's a totally different thing. `HumanoidLooksComponent`? `HumanoidAppearanceComponent`?

File: Content.Client/UserInterface/HumanoidProfileEditorPanel.cs [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+        public HumanoidCharacterProfile Profile;
+
+        public HumanoidProfileEditorPanel(ILocalizationManager localization,
+            IResourceCache resourceCache,
+            IClientPreferencesManager preferencesManager)
+        {
+            Profile = (HumanoidCharacterProfile) preferencesManager.Preferences.SelectedCharacter;
+            CharacterSlot = preferencesManager.Preferences.SelectedCharacterIndex;
+            _preferencesManager = preferencesManager;
+            _hairPickerWindow = new HairPickerWindow(resourceCache, localization);
```
Comment (DamianX) [inline Content.Client/UserInterface/HumanoidProfileEditorPanel.cs]: These are no longer used, I forgot

**Comment by DamianX** [comment]
This should be ready for review.
Not sure why the tests keep hanging.
Note the test server's `data/database.db` might have to be manually deleted once this is merged.

**Comment by PJB3005** [comment]
Tests pass for me locally, no idea why they're hanging.

File: Content.Client/GameObjects/Components/Mobs/LooksComponent.cs (hunk 2 above)
Comment (PJB3005) [reply Content.Client/GameObjects/Components/Mobs/LooksComponent.cs]: I'd definitely go with `HumanoidAppearanceComponent`

File: Content.Client/UserInterface/CharacterSetupGui.cs [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+            public readonly Button ActualButton;
+            private IEntity _previewDummy;
+
+            public CharacterPickerButton(
+                IEntityManager entityManager,
+                IClientPreferencesManager preferencesManager,
+                ButtonGroup group,
+                ICharacterProfile profile)
+            {
+                _previewDummy = entityManager.SpawnEntityAt("HumanMob_Content",
```
Comment (PJB3005) [inline Content.Client/UserInterface/CharacterSetupGui.cs]: Would it make sense to spawn a type of dummy that only has the components necessary to generate the preview? So no stuff like hunger component and whatever.

File: Content.Shared/Preferences/HumanoidCharacterAppearance.cs [hunk 5]
```diff
@@ -5,26 +5,71 @@
 namespace Content.Shared.Preferences
 {
     [Serializable, NetSerializable]
-    public class HumanoidCharacterAppearance : ICharacterAppearance
+    public readonly struct HumanoidCharacterAppearance : ICharacterAppearance
```
Comment (PJB3005) [inline Content.Shared/Preferences/HumanoidCharacterAppearance.cs]: Don't make this a struct. There's so many large structs it's way too heavy.

File: Content.Client/UserInterface/CharacterSetupGui.cs (hunk 4 above)
Comment (DamianX) [reply Content.Client/UserInterface/CharacterSetupGui.cs]: Yes, definitely

**Pull request merged**